plugins/module_utils/user.py

```diff
@@ -368,7 +368,7 @@ def user_delete(cursor, user, host, host_all, check_mode):
         hostnames = [host]
 
     for hostname in hostnames:
-        cursor.execute("DROP USER %s@%s", (user, hostname))
+        cursor.execute("DROP USER /*!50708 IF EXISTS */ /*M!100103 IF EXISTS */  %s@%s", (user, hostname))
```

Comment:
```suggestion
        try:
            cursor.execute("DROP USER IF EXISTS  %s@%s", (user, hostname))
        except:
            cursor.execute("DROP USER %s@%s", (user, hostname))
```

Reply:
@TodoInTX @rsicart sorry for the noise, i didn't sleep well, so please ignore all my comments above. This is my proposal.

Reply:
same for role

Reply:
no problem @Andersson007 :)
and thanks for helping with the review!

Personally I'm more comfortable with this solution too.

@TodoInTX what do you think about that proposal?

Reply:
@TodoInTX what do you think about the changes proposed earlier in this conversation?